dd-trace/src/main/java/datadog/trace/tracer/Tracer.java

```
@@ -115,7 +133,13 @@ void reportError(final String message, final Object... args) {
   }
 
   @Override
-  public void close() throws IOException {
+  public void finalize() {
+    Runtime.getRuntime().removeShutdownHook(shutdownCallback);
```

This can thrown an exception and generally `finalize` methods should be wrapped in try/catch because they silently eat those exceptions. Also if above call throws exception tracer will not get closed.

But overall: is this really valuable to essentially delay shutdown for the purposes of sending pending stats?